src/GitHub.VisualStudio/GitHubPackage.cs

```diff
@@ -126,11 +131,15 @@ Version VSVersion
             }
         }
 
-        tasks.Task<object> CreateService(IAsyncServiceContainer container, CancellationToken cancellationToken, Type serviceType)
+        async tasks.Task<object> CreateService(IAsyncServiceContainer container, CancellationToken cancellationToken, Type serviceType)
         {
-            AssemblyResolver.InitializeAssemblyResolver();
-            var ret = Services.ComponentModel.DefaultExportProvider.GetExportedValueOrDefault<IUIProvider>();
-            return tasks.Task.FromResult((object)ret);
+            if (serviceType == null)
+                return null;
+            string contract = AttributedModelServices.GetContractName(serviceType);
+            var cm = await GetServiceAsync(typeof(SComponentModel)) as IComponentModel;
+            if (cm == null)
+                return null;
+            return await tasks.Task.Run(() => cm.DefaultExportProvider.GetExportedValueOrDefault<object>(contract));
```

Comment:
Similar to comment above, CreateService should also be called from thread pool threads in async services case so GetExportedValueOrDefault can be called directly.